app/ts/background/popupMessageHandlers.ts

```diff
@@ -328,3 +329,33 @@ export async function changeSettings(simulator: Simulator, parsedRequest: Change
 	if (parsedRequest.data.useTabsInsteadOfPopup !== undefined) setUseTabsInsteadOfPopup(parsedRequest.data.useTabsInsteadOfPopup)
 	return await homeOpened(simulator)
 }
+
+export async function importSettings(settingsData: ImportSettings) {
+	console.log(settingsData.data.fileContents)
+	if (!isJSON(settingsData.data.fileContents)) {
+		return await sendPopupMessageToOpenWindows({
+			method: 'popup_initiate_export_settings_reply',
+			data: { success: false, errorMessage: 'Failed to read the file. It is not a valid JSOn file.' }
+		})
+	}
+	const parsed = ExportedSettings.safeParse(JSON.parse(settingsData.data.fileContents))
+	if (!parsed.success) {
+		return await sendPopupMessageToOpenWindows({
+			method: 'popup_initiate_export_settings_reply',
+			data: { success: false, errorMessage: 'Failed to read the file. It is not a valid interceptor settings file' }
+		})
+	}
+	await importSettingsAndAddressBook(parsed.value)
+	return await sendPopupMessageToOpenWindows({
+		method: 'popup_initiate_export_settings_reply',
+		data: { success: true }
+	})
+}
+
+export async function exportSettings() {
+	const exportedSettings = await exportSettingsAndAddressBook()
+	await sendPopupMessageToOpenWindows({
+		method: 'popup_initiate_export_settings',
+		data: { fileContents: JSON.stringify(ExportedSettings.serialize(exportedSettings), undefined, 4) }
```

Comment:
No bigints or Uint8Arrays will ever be serialized by this?  In my project I have these two functions to help with (de)serializing stuff that may contain bigints or Uint8Arrays:
```ts
export function jsonStringify(value: unknown, space?: string | number | undefined): string {
    return JSON.stringify(value, (_key, value) => {
		if (typeof value === 'bigint') return `0x${value.toString(16)}n`
		if (value instanceof Uint8Array) return `b'${Array.from(value).map(x => x.toString(16).padStart(2, '0')).join('')}'`
		return value
    }, space)
}
export function jsonParse(text: string): unknown {
	return JSON.parse(text, (_key: string, value: unknown) => {
		if (typeof value !== 'string') return value
		if (/^0x[a-fA-F0-9]+n$/.test(value)) return BigInt(value.slice(0, -1))
		const bytesMatch = /^b'(:<hex>[a-fA-F0-9])+'$/.exec(value)
		if (bytesMatch && 'groups' in bytesMatch && bytesMatch.groups && 'hex' in bytesMatch.groups) return hexToBytes(`0x${bytesMatch.groups['hex']}`)
		return value
	})
}
```

Reply:
serialize converts everything to dictionaries, numbers, strings ad arrays. so its all json compatible. If you want to serialize bigint, its converted to string with serialize. uint8Array's look bit funky as they become arrays of numbers, but are serializeables